sphinx_immaterial/apidoc/apidoc_formatting.py

```diff
@@ -1,524 +1,150 @@
-        default_options[name] = registered_option.default
+    # Ensure pygments uses <code> elements, for compatibility with the
+    # mkdocs-material CSS which expects that.
+    self.highlighter.formatter_args.update(wrapcode=True)
 
-    # Validate options
-    for i, (pattern, options) in enumerate(
-        pydantic.parse_obj_as(
-            List[Tuple[Pattern, Dict[str, Any]]],
-            DEFAULT_OBJECT_DESCRIPTION_OPTIONS + app.config.object_description_options,
-        )
-    ):
-        for name, value in options.items():
-            registered_option = registry.get(name)
-            if registered_option is None:
-                logger.error(
-                    "Undefined object description option %r specified for pattern %r",
-                    name,
-                    pattern.pattern,
-                )
-                continue
-            try:
-                options[name] = pydantic.parse_obj_as(
-                    registered_option.type_constraint, value
-                )
-            except Exception as e:  # pylint: disable=broad-except
-                logger.error(
-                    "Invalid value %r for object description option"
-                    " %r specified for pattern %r: %s",
-                    value,
-                    name,
-                    pattern.pattern,
-                    e,
-                )
-        if pattern.pattern == re.escape(pattern.pattern):
-            # Pattern just matches a single string.
-            options_map.setdefault(pattern.pattern, []).append((i, options))
-        else:
-            options_patterns.append((pattern, i, options))
+    # Ensure classes like `s` (used for string literals in code
+    # highlighting) aren't converted to `<s>` elements (strikethrough).
+    # Sphinx already overrides this, but for some reason due to
+    # `__init__` invocation order it gets overridden.
+    self.supported_inline_tags = set()
 
-    @functools.lru_cache(maxsize=None)
-    def get_options(domain: str, object_type: str) -> Dict[str, Any]:
-        key = f"{domain}:{object_type}"
-        matches = options_map.get(key)
-        if matches is None:
-            matches = []
-        else:
-            matches = list(matches)
-        for pattern, i, options in options_patterns:
-            if pattern.fullmatch(key):
-                matches.append((i, options))
-        matches.sort(key=lambda x: x[0])
-        full_options = default_options.copy()
-        for _, m in matches:
-            full_options.update(m)
-        full_options.update(domain=domain, object_type=object_type)
-        return full_options
 
-    app._sphinx_immaterial_get_object_description_options = get_options  # type: ignore
+def _config_inited(
+    app: sphinx.application.Sphinx, config: sphinx.config.Config
+) -> None:
+    # make code-blocks' line numbers be a separate column of a 1-row table
+    config["html_codeblock_linenos_style"] = "table"  # default is "inline"
```

Comment:
This will have to change soon (related to https://github.com/jbms/sphinx-immaterial/issues/26#issuecomment-1067465334). Sphinx v6.0 will no longer support this `html_codeblock_linenos_style` config option (which means inline line numbers for code blocks). Fortunately, the upstream CSS has support for a custom HTML attribute `data-linenos` ([originally provided by `pymdownx.highlight`](https://facelessuser.github.io/pymdown-extensions/extensions/highlight/#line-number-styles)), but that will mean monkey patching the code-blocks output to use the new `data-linenos` attribute.